From ffaf8652b13879fa3a6f8957964ecb90eb3ddf6a Mon Sep 17 00:00:00 2001
From: Anu Engineer <aengineer@apache.org>
Date: Tue, 7 Jun 2016 15:10:22 -0700
Subject: [PATCH 1605/2748] HDFS-10496. DiskBalancer: ExecuteCommand checks
 planFile in a wrong way. Contributed by Lei
 (Eddy) Xu.

(cherry picked from commit 32058f9b6d49b93ac8e99da383cb86170a16b317)

Change-Id: I58e36aae51b66cd8a1d3d1301328854688f6da3b
---
 .../diskbalancer/command/ExecuteCommand.java       |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/ExecuteCommand.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/ExecuteCommand.java
index 1f7e81f..5f3db8f 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/ExecuteCommand.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/ExecuteCommand.java
@@ -61,7 +61,7 @@ public void execute(CommandLine cmd) throws Exception {
     verifyCommandOptions(DiskBalancer.EXECUTE, cmd);
 
     String planFile = cmd.getOptionValue(DiskBalancer.EXECUTE);
-    Preconditions.checkArgument(planFile == null || planFile.isEmpty(),
+    Preconditions.checkArgument(planFile != null && !planFile.isEmpty(),
         "Invalid plan file specified.");
 
     String planData = null;
-- 
1.7.9.5

